src/mono/mono/arch/arm64/arm64-codegen.h

```diff
@@ -1761,8 +1794,42 @@ arm_encode_arith_imm (int imm, guint32 *shift)
 #define arm_neon_umull2_2d(p, rd, rn, rm) arm_neon_3dvec_opcode ((p), VREG_FULL, 0b1, SIZE_4, 0b1010, (rd), (rn), (rm))
 
 /* NEON :: 3-register same */
-#define arm_neon_3svec_opcode(p, q, u, size, opcode, rd, rn, rm) arm_neon_opcode_3reg ((p), (q), 0b00001110001000000000010000000000 | (u) << 29 | (size) << 22 | (opcode) << 12, (rd), (rn), (rm))
-
+#define arm_neon_3svec_opcode(p, q, u, size, opcode, rd, rn, rm) arm_neon_opcode_3reg ((p), (q), 0b00001110001000000000010000000000 | (u) << 29 | (size) << 22 | (opcode) << 11, (rd), (rn), (rm))
+
+// These are more generalized macros for int binary ops:
+//   width - determines if full register or its lower half is used, one of {VREG_LOW, VREG_FULL}
+//   type  - data type of vector elements, one of {TYPE_I8, TYPE_I16, TYPE_I32, TYPE_I64}
+#define arm_neon_add(p, width, type, rd, rn, rm) arm_neon_3svec_opcode ((p), (width), 0b0, (type), 0b10000, (rd), (rn), (rm))
```

Comment:
With these more generic macros, do we still need the specific ones that you added before?